### ISSUE TITLE
Reportedly, bot map locks up when starting vaporware

### ISSUE THREAD

**Issue by ioan-chera** [ioan-chera/AutoDoom#26]
It either exits with an out of memory failure, or crashes. Needs checking.

Quote by Edward: "Probably not in scope yet,  but autodoom fails to build nodes for the vaporware demo map. It runs out of memory quickly in 32bit, and some kind of array index overflow occurs in 64bit."

**Comment by ioan-chera** [comment]
Testing on a 64-bit computer, the map fails with an out-of-bounds error caused by a malformed botmap subsector. I patched it by checking if it's an invalid seg, but what I should do is investigate why the subsector was built wrongly in the first place.

The OOM error not checked yet.

**Comment by ioan-chera** [comment]
More issues observed: it seems that more metasectors than needed are added into a compound metasector. Worse, they are simple metasectors which shouldn't be at those coordinates. Normally they're blocked by the line metasectors, which are correct.

Hopefully I can check this on a simpler map, such as E2M9 or E3M8.
